flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/assigner/SimpleSplitAssigner.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.source.assigner;
+
+import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.Deque;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.apache.flink.annotation.Internal;
+import org.apache.iceberg.flink.source.split.IcebergSourceSplit;
+import org.apache.iceberg.flink.source.split.IcebergSourceSplitState;
+import org.apache.iceberg.flink.source.split.IcebergSourceSplitStatus;
+
+/**
+ * Since all methods are called in the source coordinator thread by enumerator,
+ * there is no need for locking.
+ */
+@Internal
+public class SimpleSplitAssigner implements SplitAssigner {
+
+  private final Deque<IcebergSourceSplit> pendingSplits;
+  private CompletableFuture<Void> availableFuture;
+
+  public SimpleSplitAssigner() {
+    this.pendingSplits = new ArrayDeque<>();
+  }
+
+  public SimpleSplitAssigner(Collection<IcebergSourceSplitState> assignerState) {
+    this.pendingSplits = new ArrayDeque<>(assignerState.size());
+    assignerState.stream().forEach(splitState -> pendingSplits.add(splitState.split()));
```

Comment:
Is this assuming that all of the splits are unassigned?

Reply:
that is correct. will add inline comments to explain